sbt-app/src/sbt-test/project-matrix/jvm-sandwich/build.sbt

```diff
@@ -1,14 +1,20 @@
 lazy val check = taskKey[Unit]("")
-lazy val scala3M1 = "3.0.0-M1"
+lazy val scala3 = "3.5.2"
 lazy val scala3M2 = "3.0.0-M2"
 lazy val scala213 = "2.13.4"
 
 lazy val fooApp = (projectMatrix in file("foo-app"))
   .dependsOn(fooCore)
   .settings(
     name := "foo app",
+    TaskKey[Unit]("check") := {
+      val deps = (Compile / projectDependencies).value
+      deps.foreach { dep =>
+        println(s"$dep - ${dep.crossVersion}")
```

Comment:
This prints out as
```
[info] foo-core:foo-core:0.1.0-SNAPSHOT - Binary(, )
```

indicating that the constant `crossVersion` gets lost somewhere.

Reply:
https://github.com/sbt/sbt/actions/runs/11813042758/job/32909316617?pr=7881#step:15:1173